Add project list page with live websocket support.

kialam opened the pull request:
This PR adds live web socket events to project sync within the awx projects list.
![projectslistws](https://user-images.githubusercontent.com/2293210/228994277-23284ad5-148e-4e21-87d7-ec3229962187.gif)

Cypress e2e tests:
```
Running:  awx/resources/projects.cy.ts


  projects
    ✓ can render the projects list page (23118ms)
    ✓ can navigate to project details tab (4433ms)
    ✓ can navigate to project access tab (3747ms)
    ✓ can navigate to project job templates tab (3840ms)
    ✓ can navigate to project notifications tab (3984ms)
    ✓ can navigate to project schedules tab (4005ms)
    ✓ can copy project from project details page (6162ms)
    ✓ can delete project from project details page (7986ms)
    ✓ can sync project from project details page (6228ms)
    ✓ can sync project from projects list table row kebab menu (8000ms)
    ✓ can copy project from projects list table row kebab menu (13328ms)
    ✓ can delete project from projects list table row kebab menu (12246ms)
    ✓ can delete project from projects list toolbar  (7688ms)


  13 passing (2m)
```
Cypress unit tests:
```
projects.cy.ts
  Non-empty list
    ✓ Projects list renders (745ms)
    ✓ Projects list has filters for Name, Description, Type, Created By, and Modified By (1393ms)
    ✓ Filter projects by name (3849ms)
    ✓ Filter projects by description (4218ms)
    ✓ Filter projects by created by (3940ms)
    ✓ Filter projects by modified by (3960ms)
    ✓ Filter projects by Type > Git (4537ms)
    ✓ Filter projects by Type > Subversion (4411ms)
    ✓ Filter projects by Type > Insights (4198ms)
    ✓ Filter projects by Type > Manual (4130ms)
    ✓ Filter projects by Type > Remote Archive (4008ms)
    ✓ Sync project calls API /update endpoint (1486ms)
    ✓ Create Project button is disabled if the user does not have permission to create projects (303ms)
    ✓ Sync, Copy, Edit, Delete Project button is disabled if the user does not have permission(s) (2413ms)
    ✓ Create Project button is enabled if the user has permission to create projects (291ms)
    ✓ Displays error if projects are not successfully loaded (1530ms)
  Empty list
    ✓ Empty state is displayed correctly for user with permission to create projects (511ms)
    ✓ Empty state is displayed correctly for user without permission to create projects (460ms)


18 passing (54s)
```

vidyanambiar reviewed (commented):
Looks great! Kudos on the extensive test coverage! 🎉 

I noticed that the existing AWX UI has the ability to cancel a sync once it's started, a cancel button replaces the sync button within the table row. This might be a good thing to add to the table row as well as in the kebab menu on the details page. But totally okay to add it in a separate PR I think.

kialam commented:
@vidyanambiar thank you for taking a look. I added a cancel button to the row item kebab as well as the project list toolbar (we now have bulk cancel (!)) and reworked some of the e2e tests. I think this is ready for a re-review. I believe we'll have to peel off the expandable row work (plus the work to show the last used and execution env. fields) into a separate ticket while we wait for guidance from UXD on which columns to keep.

vidyanambiar reviewed (commented):
- The non-actionable rows should float to the top of the list. Not sure why that isn't happening. I also see that issue in the Jobs list. But it works correctly in Teams. The code for that is [here](https://github.com/ansible/ansible-ui/blob/main/framework/PageDialogs/BulkConfirmationDialog.tsx#L100). We can address  it here or else in a follow-up PR if that's better.
- Minor: The confirmation message appears to be plural `Yes, I confirm that I want to cancel these project sync jobs.` even for a single cancellable job.
- The status of the project does not seem to update in the bulk confirmation dialog. It stays in the "running" state even when the table in the background has updated to show that it is now successful. This too is an issue with Jobs as well so probably ok to address them together separately.
-  Should we disable the sync button in the row action if a sync is already in progress?
![Screenshot 2023-04-06 at 12 44 55 PM](https://user-images.githubusercontent.com/43621546/230443558-199c321c-465f-4e6d-8d7a-e1837478ff0e.png)

kialam commented:
The bulk action dialog issues mentioned above are happening elsewhere and should be addressed as a separate part of this PR. Project sync is now disabled if a project has an active sync status and there is now a string for translation to handle singular and plural bulk project sync cancel.

vidyanambiar commented:
@kialam I see this [commit](https://github.com/ansible/ansible-ui/pull/357/commits/ed19039f5abf91b7b4bec76c27708117e7923768) in the commit history for disabling the sync when it is progress but I suspect some following commit is undoing that change 🤔 because it's not in the final set of file changes and the Sync button remains enabled when a sync is in progress.

vidyanambiar commented on frontend/awx/resources/projects/hooks/useProjectActions.tsx:
```diff
@@ -26,10 +35,14 @@ export function useProjectActions(options: { onProjectsDeleted: (projects: Proje
+        return t(`The project cannot be synced due to insufficient permission`);
+      }
+      return '';
+    };
```
@kialam I see this [commit](https://github.com/ansible/ansible-ui/pull/357/commits/ed19039f5abf91b7b4bec76c27708117e7923768) in the commit history for disabling the sync when it is progress but I suspect some following commit is undoing that change 🤔 because it's not in the final set of file changes and the Sync button remains enabled when a sync is in progress.

kialam replied:
I saw that too. The last commit is the offender here - as to how it snuck in ... 🐉 👀 

This also exposed a vulnerability that needs to be patched in my unit tests (I added a test to see if project sync is disabled). I'll have changes pushed on Monday when I'm back from PTO. Thanks!
